core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/replication-repmgr-Debian.yml

```diff
@@ -1,185 +1,181 @@
-  when:
-    - groups['postgresql'][1] == inventory_hostname
-    - is_node_already_registered.stdout == ""
-
-- name: Start postgresql service
-  service:
-    name: "{{ pg_service_name[ansible_os_family] }}"
-    state: started
-  when:
-    - groups['postgresql'][1] == inventory_hostname
-    - is_node_already_registered.stdout == ""
-
-- name: Restart repmgrd service
-  service:
-    name: "{{ repmgr_service_name[ansible_os_family] }}"
-    state: restarted
-  when:
-    - groups['postgresql'][1] == inventory_hostname
-    - is_node_already_registered.stdout == ""
+# For repmgr installed from Ubuntu package additional configuration is required before repmgrd is started as daemon
+- name: Set repmgr.conf file in /etc/default/repmgrd
+  replace:
+    path: /etc/default/repmgrd
+    regexp: "^#REPMGRD_CONF=\"/path/to/repmgr.conf\""
+    replace: "REPMGRD_CONF=\"{{ repmgr_config_dir[ansible_os_family] }}/repmgr.conf\""
 
-- name: Register node with repmgr
-  become_user: postgres
-  shell: "{{ repmgr_bindir[ansible_os_family] }}/repmgr standby -F -f {{ repmgr_config_dir[ansible_os_family] }}/repmgr.conf --upstream-conninfo='host={{ hostvars[groups['postgresql'][0]]['ansible_default_ipv4']['address'] }}, user={{ specification.extensions.replication.replication_user_name }}, dbname={{ specification.extensions.replication.repmgr_database }}, connect_timeout=2' --upstream-node-id 1 register"
-  when:
-    - groups['postgresql'][1] == inventory_hostname
-    - is_node_already_registered.stdout == ""
+- name: Enable repmgrd in /etc/default/repmgrd
+  replace:
+    path: /etc/default/repmgrd
+    regexp: "^[#]?REPMGRD_ENABLED=no"
+    replace: "REPMGRD_ENABLED=yes"
 
-- name: Check if node is not attached to repmgr
-  become_user: postgres
-  shell: >-
-    set -o pipefail &&
-    {{ repmgr_bindir[ansible_os_family] }}/repmgr cluster show -f {{ repmgr_config_dir[ansible_os_family] }}/repmgr.conf | grep -i 'not attached to its upstream node'
-  changed_when: false
-  register: is_node_not_attached
-  failed_when: is_node_not_attached.rc == 2
-  args:
-    executable: /bin/bash
-  when:
-    - groups['postgresql'][1] == inventory_hostname
-
-- name: Rejoin node with repmgr
-  become_user: postgres
-  shell: "{{ repmgr_bindir[ansible_os_family] }}/repmgr standby -f {{ repmgr_config_dir[ansible_os_family] }}/repmgr.conf follow"
-  when:
-    - groups['postgresql'][1] == inventory_hostname
-    - not is_node_not_attached.stdout == ""
-
-# All
-- name: Restart repmgrd service
-  service:
-    name: "{{ repmgr_service_name[ansible_os_family] }}"
-    state: restarted
+# --- Configure primary node ---
+
+- name: Tasks to run only on primary node
+  when: groups.postgresql[0] == inventory_hostname
```

Comment:
Again this is not in scope of this PR, but I think we talked about this somewhere before. This is a naive asumption that we can rely on ordering of inventory groups. Especially in AWS where we miss-use autocaling groups and in the end make Epiphany `un-scallable` in the process. :(

Reply:
I created issue #1846 to solve that.